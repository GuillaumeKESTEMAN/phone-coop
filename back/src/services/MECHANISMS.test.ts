import initMechanisms, { FAKE_MECHANISM } from './MECHANISMS';
import { BEARER as BEARER_MECHANISM } from 'http-auth-utils';

describe('MECHANISMS', () => {
  const log = jest.fn();

  beforeEach(() => {
    log.mockReset();
  });

  it('should only include bearer', async () => {
    const MECHANISMS = await initMechanisms({
      ENV: {},
      log,
    });

    expect(MECHANISMS).toEqual([BEARER_MECHANISM]);

    expect({
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
    }).toMatchSnapshot();
  });

  it('should also include fake on debugging', async () => {
    const MECHANISMS = await initMechanisms({
      ENV: { DEV_MODE: '1' },
      log,
    });

    expect(MECHANISMS).toEqual([BEARER_MECHANISM, FAKE_MECHANISM]);

    expect({
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
    }).toMatchSnapshot();
  });
});
