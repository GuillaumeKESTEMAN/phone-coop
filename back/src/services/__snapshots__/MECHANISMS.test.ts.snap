// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MECHANISMS should also include fake on debugging 1`] = `
Object {
  "logCalls": Array [
    Array [
      "debug",
      "üîß - Initializing auth mechanisms",
    ],
    Array [
      "warning",
      "‚ö†Ô∏è - Using fake auth mechanism!",
    ],
  ],
}
`;

exports[`MECHANISMS should only include bearer 1`] = `
Object {
  "logCalls": Array [
    Array [
      "debug",
      "üîß - Initializing auth mechanisms",
    ],
  ],
}
`;
